Final touches to pages (#226)
MystenLabs/suins-contracts: documentation/pages/move-registry/managing-package-info.mdx

import Beta from '../../components/beta'
import { Callout } from 'nextra/components';

<Beta/>

# Creating/Maintaining `PackageInfo` object

<Callout type="warning">

>     These examples assume that you use the Typescript Plugin, as described in the  [`Typescript SDK`](/move-registry/tooling/typescript-sdk) section.

Reviewer: ```suggestion
    These examples assume that you use the MVR plugin for the Sui Typescript SDK, as described in the  [`Typescript SDK`](/move-registry/tooling/typescript-sdk) section.
```
I think wording like this makes it a little clearer what it is? I think it's technically correct?

    arguments: [transaction.object('<Your UpgradeCap (by value or object id)>')],
});

// We also need to create the visual representation of our "info" object.
// You can also call `@mvr/metadata::display::new` instead, 
// that allows customizing the colors of your metadata object!
const display = transaction.moveCall({
    target: `@mvr/metadata::display::default`,
    arguments: [
        transaction.pure.string('<Add a name to easily identify your package. This is not your MVR name.>')
    ],
});

// Set that display object to our info object.
transaction.moveCall({
    target: `@mvr/metadata::package_info::set_display`,
    arguments: [transaction.object(packageInfo), display],
});

// transfer the `PackageInfo` object to a safe address. 
transaction.moveCall({
    target: `@mvr/metadata::package_info::transfer`,
    arguments: [transaction.object(packageInfo), transaction.pure.address('<Your safe address>')],
});

// .. you can do any other actions like setting the source code info, in the same PTB.
```

## Adding source code information

You can add source code information on your `PackageInfo` object by the following setup.

> To update the source for a version, you should first call 

Reviewer: ```suggestion
To update the source for a version, first call 
```

`@mvr/metadata::package_info::unset_git_versioning` to remove the existing information.

```typescript
const transaction = new Transaction();
// adding source code info:

// 1. Create our `GitInfo` object.
const git = transaction.moveCall({
    target: `@mvr/metadata::git::new`,
    arguments: [
        transaction.pure.string('<Your git repository, e.g. `https://github.com/mystenlabs/mvr`>'),
        transaction.pure.string('<Your git subdirectory, e.g. `packages/mvr`>'),
        transaction.pure.string('<Your git commit hash or tag, e.g. `636d22d6bc4195afec9a1c0a8563b61fc813acfc`>'),
    ],
});

transaction.moveCall({
    target: `@mvr/metadata::package_info::set_git_versioning`,
    arguments: [
        transaction.object(`<Your PackageInfo object>`),
        transaction.pure.u64(`<The version number for the given source code (e.g. 1)>`),
        git,
    ],
});
```

> ## Removing / updating source code information

Reviewer: ```suggestion
## Removing or updating source code information
```
I'm just not a fan of the slash except in code so disregard suggestion if you wanna.

```typescript
const transaction = new Transaction();

transaction.moveCall({
    target: `@mvr/metadata::package_info::unset_git_versioning`,
    arguments: [
        transaction.object(`<Your PackageInfo object>`),
        transaction.pure.u64(`<The version number for the given source code (e.g. 1)>`),
    ],
});
// .. you could then follow the steps above to add the new source code info for that version.
``` 

## Tranferring the `PackageInfo` object

You can transfer the `PackageInfo` object to another address (or object) by the following setup. If you want to 

> build custom access-control for your object, you can utilize TTO (Transfer-to-Object).

Reviewer: ```suggestion
build custom access control for your object, you can utilize a transfer to object workflow (see [Transfer to Object](https://docs.sui.io/concepts/transfers/transfer-to-object) on the Sui documentation site for more information).
```
